### PR TITLE
feat: Add search to ssr cohorts view

### PR DESCRIPTION
**Pull request by connoratrug**
Add SearchInput to molgenis component lib
Use searchInput in cohorts view

_additional fixes:_ 
- components lib index page; use current bootstrap version
- components client.js; use safe methods for checking has own prop

### CODE REVIEW

File: apps/molgenis-components/src/components/forms/BaseInput.vue [hunk 1]
```diff
@@ -16,6 +16,10 @@ export default {
       type: String,
       required: false
     },
+    placeholder: {
+      type: String,
+      required: false
+    },
     helpText: {
```
Comment (mswertz) [inline apps/molgenis-components/src/components/forms/BaseInput.vue:23]: what is this? If this is the 'description' in emx2 terminology then please please call this 'description'. I want this to be a tightly coupled framework to emx2 terminology where possible to make it easier to understand as a developer.

File: apps/molgenis-components/src/client/client.js [hunk 2]
```diff
@@ -95,9 +95,9 @@ const fetchTableData = async (
 ) => {
   const url = graphqlURL ? graphqlURL : "graphql";
   const limit =
-    properties && properties.hasOwnProperty('limit') ? properties.limit : 20;
+    properties && Object.prototype.hasOwnProperty.call(properties, 'limit') ? properties.limit : 20;
```
Comment (mswertz) [inline apps/molgenis-components/src/client/client.js:98]: why this change?

File: apps/molgenis-components/src/docs/forms/InputSearch.vue [hunk 3]
```diff
@@ -0,0 +1,31 @@
+<template>
+  <demo-item id="input-search" label="Search Input (extends BaseInput)">
```
Comment (mswertz) [inline apps/molgenis-components/src/docs/forms/InputSearch.vue:2]: why do we have a demo-item here? Is this some magic you use elsewhere?

Comment (mswertz) [inline apps/molgenis-components/src/docs/forms/InputSearch.vue:2]: I thought you find a way to keep the 'docs' with the component it is documenting. Now we have to maintain two files. I really would like to prevent that.

File: apps/nuxt-ssr/pages/_schema/ssr-catalogue/cohorts/index.vue [hunk 4]
```diff
@@ -2,6 +2,7 @@
   <div class="container mt-3">
     <h1>Cohorts</h1>
     <div class="pb-1">
+      <input-search id="search" v-model="search"></input-search>
```
Comment (mswertz) [inline apps/nuxt-ssr/pages/_schema/ssr-catalogue/cohorts/index.vue:5]: I really prefer to use InputSearch instead of input-search.

File: apps/nuxt-ssr/pages/_schema/ssr-catalogue/cohorts/index.vue [hunk 5]
```diff
@@ -28,24 +29,30 @@
 </template>
 
 <script>
-import cohortsQuery from "../../../../store/gql/cohorts.gql";
```
Comment (mswertz) [inline apps/nuxt-ssr/pages/_schema/ssr-catalogue/cohorts/index.vue:31]: I have discussed with Joris and Max to go back to double quotes to not need this random format changes. Change not merge unfortunately.

File: apps/nuxt-ssr/pages/_schema/ssr-catalogue/cohorts/index.vue [hunk 6]
```diff
@@ -28,24 +29,30 @@
...
+  components: {Pagination, InputSearch},
   data() {
     return {
       cohorts: [],
+      search:
+        this.$route.query.search === undefined ? '' : this.$route.query.search,
       count: 0,
       page: 1,
-      limit: 20,
+      limit: 20
```
Comment (mswertz) [inline apps/nuxt-ssr/pages/_schema/ssr-catalogue/cohorts/index.vue:44]: same for trailing ','

**Review by mswertz — CHANGES_REQUESTED** [review]
see comments, some of it might be simply discussion.

File: apps/molgenis-components/src/components/forms/BaseInput.vue (hunk 1 above)
Comment (connoratrug) [reply apps/molgenis-components/src/components/forms/BaseInput.vue:23]: placeholder is a html attr, name a html for name, a label is a html label , the id is a html id.

File: apps/molgenis-components/src/components/forms/InputSearch.vue [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+    <input
+      :id="id"
+      :ref="id"
+      v-model="input" 
+      type="text"
+      class="form-control"
+      :aria-describedby="id"
+      placeholder="Search"
+    />
+    <small
```
[the remark this answers is not in the document]
Comment (connoratrug) [reply apps/molgenis-components/src/components/forms/InputSearch.vue:15]: please describe 'this part'

File: apps/molgenis-components/src/client/client.js (hunk 2 above)
Comment (connoratrug) [reply apps/molgenis-components/src/client/client.js:98]: https://learn.snyk.io/lessons/prototype-pollution/javascript/

File: apps/molgenis-components/src/docs/forms/InputSearch.vue (hunk 3 above)
Comment (connoratrug) [reply apps/molgenis-components/src/docs/forms/InputSearch.vue:2]: commit unclear , please elaborate

File: apps/nuxt-ssr/pages/_schema/ssr-catalogue/cohorts/index.vue (hunk 4 above)
Comment (connoratrug) [reply apps/nuxt-ssr/pages/_schema/ssr-catalogue/cohorts/index.vue:5]: i prefer input-search

File: apps/nuxt-ssr/pages/_schema/ssr-catalogue/cohorts/index.vue (hunk 5 above)
Comment (connoratrug) [reply apps/nuxt-ssr/pages/_schema/ssr-catalogue/cohorts/index.vue:31]: please update lint settings

File: apps/nuxt-ssr/pages/_schema/ssr-catalogue/cohorts/index.vue (hunk 6 above)
Comment (connoratrug) [reply apps/nuxt-ssr/pages/_schema/ssr-catalogue/cohorts/index.vue:44]: please update lint settings

File: apps/molgenis-components/src/docs/forms/InputSearch.vue (hunk 3 above)
Comment (connoratrug) [reply apps/molgenis-components/src/docs/forms/InputSearch.vue:2]: this would mean spending more time on the showCase setup , we discussed this and decided on not doing this

**Comment by connoratrug** [comment]
[#8959 SSR search, pages, filters](https://trello.com/c/o0EAlZKR/434-8959-ssr-search-pages-filters)

**Comment by connoratrug** [comment]
related: #1108
